lib/platform-stats.js

```diff
@@ -15,15 +15,28 @@ export const updatePlatformStats = async (pgClient, honestMeasurements) => {
  * @param {import('./preprocess').Measurement[]} honestMeasurements
  */
 export const updateDailyStationStats = async (pgClient, honestMeasurements) => {
-  // TODO: when we add more fields, we will update the ON CONFLICT clause
-  // to update those fields, and we won't just use a Set for the stationIds
-  // which currently removes all granular measurement details
-  const stationIds = [...new Set(honestMeasurements.map(m => m.stationId))]
+  const stationIds = []
+  const counts = []
+  const honestMeasurementCounts = new Map()
+  for (const m of honestMeasurements) {
+    honestMeasurementCounts.set(m.stationId, (honestMeasurementCounts.get(m.stationId) || 0) + 1)
+  }
+
+  for (const [stationId, count] of honestMeasurementCounts.entries()) {
+    stationIds.push(stationId)
+    counts.push(count)
+  }
+
   debug('Updating daily station stats, unique_count=%s', stationIds.length)
 
+  // Coalesce to 0 to avoid null + count = null
+  // Use EXCLUDED to allow incrementing the count
+  // Parameterized stationId to prevent SQL injection
   await pgClient.query(`
-    INSERT INTO daily_stations (station_id, day)
-    SELECT unnest($1::text[]), now()
-    ON CONFLICT (station_id, day) DO NOTHING
-  `, [stationIds])
+    INSERT INTO daily_stations (station_id, day, honest_measurement_count)
+    VALUES (unnest($1::text[]), now(), unnest($2::int[]))
+    ON CONFLICT (station_id, day) DO UPDATE
+    SET honest_measurement_count = COALESCE(daily_stations.honest_measurement_count, 0)
+                                    + EXCLUDED.honest_measurement_count
+  `, [stationIds, counts])
```

Comment:
```suggestion
  `, [
    Array.from(honestMeasurementCounts.keys()),
    Array.from(honestMeasurementCounts.values())
  ])
```

Reply:
I find it a bit difficult to understand what exactly is stored in the data structure called "honestMeasurementCounts".
- IIUC, we store counts per station instance; let's encode that in the name.
- I don't think it's important to mention that we are counting honest measurements only because this function does not have access to any other measurements.

A few proposed names for inspiration:
- `measurementCountsPerStation`
- `perStationStats` (assuming we will be adding more per-station fields in the future)
- `perStationMeasurementCounts`
- `stationMeasurementCounts`

Reply:
SGTM. Are we okay with the column name being honest_measurement_count still?

Reply:
Good point! Let me provide more context.

When spark-evaluate (and voyager-evaluate) downloads the measurements committed to the smart contract, we get a list of all measurements submitted for the given round. We don't know yet which measurements were performed by honest nodes and which were fraudulent.

One of the steps in the evaluation pipeline is to run various fraud-detection algorithms to detect suspicious measurements that should not be rewarded.

The outcome is a list of all measurements we consider honest and which should be rewarded.

Now, in our private dashboard, we often want to look at statistics about both sets for measurements (all measurements, honest measurements) to spot anomalies that could indicate a bug in our fraud detection.

When we started the work on the public dashboard, we realised that we must always use only honest measurements to calculate the public stats because that's the only dataset we trust enough.

However, when reading the spark-evaluate code, I think it's useful to be reminded that the input to these calculations is the set of honest measurements, that's why I used the name `honestMeasurements` instead of simply `measurements`.

When it comes to the database schema, we don't need to keep this distinction IMO, as the presentation layer (API & Dashboards) does not differentiate between "all" vs "honest" measurements.

So let's use `measurement_count` or even `measurements` as the SQL column name.

Reply:
Understood! Are we certain we'll never want to show a distinction between "all" vs "honest" measurements? I think a while ago we had briefly discussed doing this.

Reply:
> Understood! Are we certain we'll never want to show a distinction between "all" vs "honest" measurements? I think a while ago we had briefly discussed doing this.

I can't rule that out for 100%. OTOH, all other tables and columns don't encode "honest" in their names.

Oh, I think I see what you mean. Here, we are specifically talking about the number of measurements, so it makes sense to be more explicit about "honest" vs "all" measurements.

OK, let's keep "honest" in the SQL column name, but change it to "accepted". I feel it better describes the intent and does not have any negative connotations of accusing the participant of doing something wrong.